cmd/clusterctl/cmd/version_checker.go

```diff
@@ -139,28 +148,43 @@ New clusterctl version available: v%s -> v%s
 
 func (v *versionChecker) getLatestRelease() (*ReleaseInfo, error) {
 	log := logf.Log
+
+	// Try to get latest clusterctl version number from the local state file.
+	// NOTE: local state file is ignored if older than 1d.
 	vs, err := readStateFile(v.versionFilePath)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to read version state file")
 	}
+	if vs != nil {
+		return &vs.LatestRelease, nil
+	}
 
-	// if there is no release info in the state file, pull latest release from github
-	if vs == nil {
-		release, _, err := v.githubClient.Repositories.GetLatestRelease(context.TODO(), "kubernetes-sigs", "cluster-api")
-		if err != nil {
-			log.V(1).Info("⚠️ Unable to get latest github release for clusterctl")
-			// failing silently here so we don't error out in air-gapped
-			// environments.
-			return nil, nil //nolint:nilerr
+	// Try to get latest clusterctl version number from go modules.
+	ri, err := v.goproxyGetLatest()
```

Comment:
Does `ri` mean release info? If so, could we rename some of the variables to indicate what they're used for?